### PR TITLE
Prefer PETSc dof_id sizes over implicitly set sizes

### LINKED ISSUE #2060: Detect dof-id-bytes
Right now libMesh will thrown an error if I compile PETSc with 64bit indices and try to use `--with-dof-id-bytes=4` (or vice versa).

It would be great if we didn't need to specify "dof-id-bytes" at all... and just have it be autodetected to use the same thing PETSc is using.

### PR DESCRIPTION
**Pull request by roystgnr**
If the user doesn't specify otherwise then we can default to PETSc's index size rather than default to 4.

This should resolve #2060

### CODE REVIEW

File: bootstrap [hunk 1]
```diff
@@ -35,7 +35,7 @@ function check_autotool_prerequisites ()
 {
     # prerequisite versions
     automake_major=1
-    automake_minor=15
+    automake_minor=16
```
Comment (jwpeterson) [inline bootstrap]: Nice catch, I'll make a note to do this the next time we change autotools versions.

**Comment by roystgnr** [comment]
@pbauman, any idea what's going on with that regression/ozone_flame_antioch_kinetics_theory.sh divergence?  I'm trying to reproduce the libMesh example-failure-with-debug-PETSc now, but I'm not even sure where to begin on that GRINS failure.

**Comment by roystgnr** [comment]
Anyway, going to ignore the unrelated CI problems and merge.

**Comment by pbauman** [comment]
> any idea what's going on with that regression/ozone_flame_antioch_kinetics_theory.sh divergence?

No not sure what that's about. Will have a look at it tomorrow.

**Comment by friedmud** [comment]
Sweet!  Thanks for implementing this @roystgnr and @jwpeterson !  Should be a large help!

**Pull request merged**